codec-classes-quic/src/main/java/io/netty/incubator/codec/quic/QuicClientSessionCache.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright 2021 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.incubator.codec.quic;
+
+
+import io.netty.util.AsciiString;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+final class QuicClientSessionCache {
```

Comment:
This cache is currently basically unbound... We should ensure we enforce some bound. Beside this we should also ensure we remove expired sessions after some time. This is to mimic what BoringSSL is doing internally:

https://commondatastorage.googleapis.com/chromium-boringssl-docs/ssl.h.html#SSL_SESS_CACHE_NO_AUTO_CLEAR